src/Common/LRUCache.h

```diff
@@ -295,28 +350,42 @@ class LRUCache
                 cells.erase(it);
                 throw;
             }
+
+            if (!removeOverflow())
+            {
+                // overflow is caused by inserting this element.
+                queue.erase(cell.queue_iterator);
+                cells.erase(it);
+                return false;
+            }
         }
         else
         {
+            if (!evict_policy.canRelease(cell.value))
+                return false;
+            if (value_weight > cell.size && !removeOverflow(value_weight - cell.size))
+                return false;
+            evict_policy.release(cell.value); // release the old value. this action is empty in default policy.
```

Comment:
Also I am worried that `release` can throw exception...
But not sure should we catch it here or just make sure in `release` method that it never throws.

Reply:
Ok I the only solution is to make `release` and `canRelease` as `noexcept`. Because it is controversial if or how we should process exceptions from `release` for old keys.